android/src/main/java/com/dooboolab/fluttersound/FlutterSoundPlugin.java

```diff
@@ -28,6 +28,31 @@
 import io.flutter.plugin.common.PluginRegistry;
 import io.flutter.plugin.common.PluginRegistry.Registrar;
 
+// SDK compatibility
+// -----------------
+
+class sdkCompat {
+  static final int AUDIO_ENCODER_VORBIS = 6;  // MediaRecorder.AudioEncoder.VORBIS added in API level 21
+  static final int AUDIO_ENCODER_OPUS   = 7;  // MediaRecorder.AudioEncoder.OPUS   added in API level 29
+  static final int OUTPUT_FORMAT_OGG    = 11; // MediaRecorder.OutputFormat.OGG    added in API level 29
+  static final int VERSION_CODES_M      = 23; // added in API level 23
+
+  static int checkRecordPermission(Registrar reg) {
+    if (Build.VERSION.SDK_INT >= sdkCompat.VERSION_CODES_M) {// Before Marshmallow, record permission was always granted.
+      Activity activity = reg.activity();
+      if (reg.context().checkCallingOrSelfPermission(Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED) {
+        ActivityCompat.requestPermissions(activity, new String[]{Manifest.permission.RECORD_AUDIO,}, 0);
+        if (reg.context().checkCallingOrSelfPermission(Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED)
+          return PackageManager.PERMISSION_DENIED;
+      }
+    }
+    return PackageManager.PERMISSION_GRANTED;
+  }
+  protected static void setDataSource(AudioModel model, byte[] dataBuffer) {
```

Comment:
What is this method for?

Reply:
This is an error. This function is not used anymore. I tried to allow "PlayForBuffer()" to be compatible with very old Android SDK but I was not successful.

I am removing this function which is not necessary anymore.